homeassistant/components/axis/camera.py

```diff
@@ -35,27 +30,24 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
     if not device.option_camera:
         return
 
-    config = {
-        CONF_NAME: config_entry.data[CONF_NAME],
-        CONF_USERNAME: config_entry.data[CONF_USERNAME],
-        CONF_PASSWORD: config_entry.data[CONF_PASSWORD],
-        CONF_MJPEG_URL: AXIS_VIDEO.format(
-            host=config_entry.data[CONF_HOST], port=config_entry.data[CONF_PORT],
-        ),
-        CONF_STILL_IMAGE_URL: AXIS_IMAGE.format(
-            host=config_entry.data[CONF_HOST], port=config_entry.data[CONF_PORT],
-        ),
-        CONF_AUTHENTICATION: HTTP_DIGEST_AUTHENTICATION,
-    }
-    async_add_entities([AxisCamera(config, device)])
+    async_add_entities([AxisCamera(device)])
 
 
 class AxisCamera(AxisEntityBase, MjpegCamera):
     """Representation of a Axis camera."""
 
-    def __init__(self, config, device):
+    def __init__(self, device):
         """Initialize Axis Communications camera component."""
         AxisEntityBase.__init__(self, device)
+
+        config = {
+            CONF_NAME: device.config_entry.data[CONF_NAME],
+            CONF_USERNAME: device.config_entry.data[CONF_USERNAME],
+            CONF_PASSWORD: device.config_entry.data[CONF_PASSWORD],
+            CONF_MJPEG_URL: self.mjpeg_source,
+            CONF_STILL_IMAGE_URL: self.image_source,
+            CONF_AUTHENTICATION: HTTP_DIGEST_AUTHENTICATION,
+        }
         MjpegCamera.__init__(self, config)
```

Comment:
It would be good if axis would implement its own camera interface in a library so we don't depend on the mjpeg camera platform here. That's not according to our standards.

Reply:
Ok. Didn't know that. 2.5 year old code :).Do you have an example I can reference?

Reply:
Since axis is using the mjpeg camera you could just take the parts that you use (digest authentication) from that platform and put in a library.

Reply:
Great, I will do that for a future PR